crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -47,6 +49,38 @@ pub enum Visibility {
     Visible,
 }
 
+#[derive(Error, Debug)]
+#[error("could not convert Visibility::Inherited to bool")]
+pub struct InheritedVisibilityToBoolConversionError;
```

Comment:
Usually, these kinds of errors are `enum`s where the specific variant points towards the more exact reason. That way, we can add more variants in the future and increase specificity without breaking changes. 
The error message could also be improved if said variants held the actual value we tried to convert. 
E.g. a `VariantNotSupported(Visibility)` would be able to show an error like "The variant {:?} cannot be converted to a bool". As-is, a user running into this error will have to manually debug or look at the source to see the actual issue.

I won't block on this though; rest seems sensible :)

Reply:
Hi there! Is that an implementation You suggested?
```rs
impl Display for Visibility{
    fn fmt(&self,f: &mut fmt::Formatter) -> fmt::Result{
        write!(f, "{}", self)
    }
}

/// Enum of errors that could occur during conversion to [`bool`]
/// `VariantNotSupported(Visibility)` Provided variant cannot be converted to [`bool`]
#[non_exhaustive]
#[derive(Error, Debug)]
pub enum VisibilityToBoolConversionError{
    #[error("The variant `{0:?}` cannot be converted to a bool")]
    VariantNotSupported(Visibility),
}
```

Reply:
@Stanleeeeey basically, yeah :) 
The second doc comment on the enum is probably there by mistake and your `Display` is recursive, looks perfect otherwise ;)